connect-button/src/main/java/com/ifttt/connect/ui/ConnectButton.java

```diff
@@ -434,6 +438,16 @@ public Builder setInviteCode(String inviteCode) {
                 return this;
             }
 
+            /**
+             * An optional configuration to render the connect button on a dark background
+             * Call this method while building the configuration if your activity has a dark background color
+             * @return The Builder object itself for chaining.
+             */
+            public Builder setOnDarkBackground(boolean onDarkBackground) {
```

Comment:
What is the reasoning behind adding this as a configuration attribute, instead of a ConnectButton attribute? Do we think it should be something that the user of this class to re-configure if they want to change this flag?

Reply:
My reasoning for adding it as a configuration attribute is:
- it can be set while we build the `ConnectButton.Configuration` and doesn't need to be called separately i.e. `connectButton.setOnDarkBackground(true)`

Reply:
If you look at the Configuration class, it's mostly around the functionality side of things for the ConnectButton, instead of the UI/rendering. It is designed to be called only once when the button is being set up (it reassigns the user email and other information). 

On the other hand, the BaseConnectButton's setOnDarkBackground is implemented in a way that it can be called multiple times and different places, the change should be applied to the button UI right away.

My concern for mixing these two different types of attributes would cause confusion, and can make it harder to use the method if the caller just want to reset the UI but keep everything else for the button unchanged.

Reply:
The last point is a valid concern, but another thing I forgot to mention earlier is that the  `setOnDarkBackground` method of `BaseConnectButton` class must be called before `setConnection` is called for the UI changes to take effect.

https://github.com/IFTTT/ConnectSDK-Android/blob/e2bd5371d36eb078baea9234da1f6fbbf5bc3913/connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java#L277

So the caller cannot really reset the UI without calling `setUp` again. Bundling UI flag with configuration builder is a way to make sure the flag is always set before any UI is rendered. 

Open to changing this, and adding a comment to make sure that the caller must invoke  `connectButton.setOnDarkBackground(true)` before setUp if they want the UI changes to be effective.

Reply:
Talked irl and refactored the code so that `setConnection` doesn't depend on the background flag and the flag can exist as a button attribute outside of configuration.